src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+            The view coordinates (i.e. minimum viewable angle) for each
+            satellite in a numpy array. If None, view coordinates will be
+            evenly distributed across the viewable range.
+        num_requests : int, default = 48
+            The number of requests to generate.
+        turn_rate : float, default = 2
+            How quickly each satellite may reorient its view angle.
+        solution_criteria : float, default = 0.99
+            The target for a successful solution. The solver will stop
+            looking for a better schedule if the specified fraction of
+            requests is satisfied.
+        """
+        super(SatelliteScheduleProblem,
+              self).__init__(num_agents=num_satellites,
+                             num_tasks=num_requests,
+                             sat_cutoff=solution_criteria,
+                             seed=seed)
+        self.num_satellites = self.num_agents
+        self.num_requests = self.num_tasks
+
+        self.view_height = view_height
+        if view_coords is None:
+            self.view_coords = np.linspace(0 - view_height / 2,
+                                           1 - view_height / 2,
+                                           num_satellites)
+        else:
+            self.view_coords = view_coords
+        self.agent_attrs = list(zip([view_height] * num_satellites,
+                                    self.view_coords))
+        self.satellites = self.agent_ids
+        self.turn_rate = turn_rate
+        self.requests = None
+        self.qubo_problem = None
+        self.generate_requests(requests)
+
+    def generate_requests(self, requests=None):
+        """ Generate a random set of requests in the 2D plane. """
+        if requests is not None:
+            self.requests = requests
+        else:
+            np.random.seed(self.random_seed)
+            self.requests = np.random.random((self.num_requests, 2))
+            order = np.argsort(self.requests[:, 0])
+            self.requests = self.requests[order, :]
+        self.task_attrs = self.requests.tolist()
+
+    def is_node_valid(self, sat_id, req_id):
+        """ Return whether the request is visible to the satellite. """
+        view_height = self.agent_attrs[sat_id][0]
+        satellite_y_coord = self.agent_attrs[sat_id][1]
+        request_y_coord = self.task_attrs[req_id][1]
+        lower_bound = satellite_y_coord
+        upper_bound = satellite_y_coord + view_height
+        return lower_bound <= request_y_coord <= upper_bound
+
+    def is_req_reachable(self, n1, n2):
+        nodes = self.graph.nodes
+        n1_req_coords = nodes[n1]["task_attr"]
+        n2_req_coords = nodes[n2]["task_attr"]
+        delta_x = abs(n1_req_coords[0] - n2_req_coords[0])
+        delta_y = abs(n1_req_coords[1] - n2_req_coords[1])
+        return self.turn_rate * delta_x >= delta_y
+
+    def is_edge_conflicting(self, node1, node2):
```

Comment:
Inhowfar does this differ from is_edge_conflict of the Scheduler class above?

Reply:
A lot. Here we check the reachability of the next request from a previous request in the specific context of a Satellite Scheduling Problem. The parent method is generic.

Reply:
Would appreciate if you could add that explanation to a docstring!